utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/TestResultSinkFactory.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.engine.flink.util.test
+
+import org.apache.flink.api.common.functions.FlatMapFunction
+import org.apache.flink.streaming.api.functions.sink.SinkFunction
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.process.{Sink, SinkFactory}
+import pl.touk.nussknacker.engine.flink.api.process.{BasicFlinkSink, FlinkLazyParameterFunctionHelper}
+import pl.touk.nussknacker.engine.flink.util.test.TestResultSinkFactory.TestResultSink
+import pl.touk.nussknacker.engine.testmode.TestRunId
+
+import java.util.concurrent.ConcurrentSkipListMap
+
+class TestResultSinkFactory(runId: TestRunId) extends SinkFactory {
+
+  @MethodToInvoke
+  def create(@ParamName("value") value: LazyParameter[AnyRef]): Sink =
+    new TestResultSink(value, runId)
+
+}
+
+object TestResultSinkFactory {
+
+  private final case class SinkId(runId: TestRunId) extends Comparable[SinkId] {
```

Comment:
Do we need extra VO? Maybe it's possible to use just TestRunId?

Reply:
as we agreed:
1. the SinkId was removed
2. TestRunId implements Comparable
3. ResultsCollectingListenerHolder uses ConcurrentSkipListMap (synchronized was removed)

Reply:
There is still normal Map with synchronized used at `TestExtensionsHolder`.